### ISSUE TITLE
configure_file assertion

### ISSUE THREAD

**Issue by legeana** [mesonbuild/meson#6974]
**Describe the bug**
Assertion error in configure_file().
Context: https://github.com/mesonbuild/libopenjp2/pull/1

```
$ mesonwrap review libopenjp2 --pull-request=1 --approve
Inspecting project libopenjp2, branch 2.3.1.
Repo name valid: ☑
Branch name valid: ☑
Target branch is not master: ☑
Has readme.txt: ☑
Has LICENSE.build: ☑
Has upstream.wrap: ☑
Repo contains only buildsystem files: ☑
upstream.wrap has directory: ☑
upstream.wrap directory has no path separators: ☑
upstream.wrap has source_url: ☑
upstream.wrap has source_filename: ☑
upstream.wrap source_filename has no path separators: ☑
upstream.wrap has source_hash: ☑
upstream.wrap has source_url with version substring: ☑
Download URL works: ☑
Hash matches: ☑
upstream.wrap directory 'openjpeg-2.3.1' exists: ☑
Patch merges with source: ☑
The Meson build system
Version: 0.54.0
Source dir: /tmp/tmp4yri2fc_/src/openjpeg-2.3.1
Build dir: /tmp/tmp4yri2fc_/bin
Build type: native build
Project name: openjp2
Project version: 2.3.1
C compiler for the host machine: cc (gcc 9.3.0 "cc (Arch Linux 9.3.0-1) 9.3.0")
C linker for the host machine: cc ld.bfd 2.34
C++ compiler for the host machine: c++ (gcc 9.3.0 "c++ (Arch Linux 9.3.0-1) 9.3.0")
C++ linker for the host machine: c++ ld.bfd 2.34
Host machine cpu family: x86_64
Host machine cpu: x86_64
Library m found: YES
Run-time dependency threads found: YES
Checking for function "clock_gettime" : YES 
wx-config found: YES (/usr/bin/wx-config) 3.0.4
Run-time dependency WxWidgets found: YES 3.0.4
Found pkg-config: /usr/bin/pkg-config (1.6.3)
Run-time dependency zlib found: YES 1.2.11
Run-time dependency libpng found: YES 1.6.37
Message: STATUS: Your system seems to have a png lib available, we will use it to generate PNG lib
Found CMake: /usr/bin/cmake (3.17.0)
Run-time dependency libtiff found: NO (tried pkgconfig and cmake)
Run-time dependency libjpeg found: YES 2.0.4
Has header "assert.h" : YES 
Has header "ctype.h" : YES 
Has header "dlfcn.h" : YES 
Has header "fcntl.h" : YES 
Has header "float.h" : YES 
Has header "inttypes.h" : YES 
Has header "ieeefp.h" : NO 
Has header "io.h" : NO 
Has header "limits.h" : YES 
Has header "malloc.h" : YES 
Has header "math.h" : YES 
Has header "memory.h" : YES 
Has header "search.h" : YES 
Has header "stdarg.h" : YES 
Has header "stdint.h" : YES 
Has header "stdlib.h" : YES 
Has header "stdio.h" : YES 
Has header "string.h" : YES 
Has header "strings.h" : YES 
Has header "sys/stat.h" : YES 
Has header "sys/time.h" : YES 
Has header "sys/types.h" : YES 
Has header "time.h" : YES 
Has header "unistd.h" : YES 
Has header "windows.h" : NO 
Has header "winsock2.h" : NO 
Checking for function "getopt" : YES 
Checking for function "isascii" : YES 
Checking for function "memmove" : YES 
Checking for function "memset" : YES 
Checking for function "mmap" : YES 
Checking for function "strcasecmp" : YES 
Checking for function "strchr" : YES 
Checking for function "strrchr" : YES 
Checking for function "strstr" : YES 
Checking for function "strtol" : YES 
Checking for function "strtoul" : YES 
Checking for function "strtoull" : YES 
Checking for function "snprintf" : YES 
Checking for size of "signed int" : 4
Checking for size of "signed long" : 8
Checking for size of "signed long long" : 8
Checking for size of "signed short" : 2
Checking for size of "unsigned int" : 4
Checking for size of "unsigned long" : 8
Checking for size of "unsigned long long" : 8
Checking for size of "unsigned short" : 2
Checking for size of "unsigned char*" : 8
Checking for size of "ssize_t" : 8
Checking for function "floor" : YES 
Checking for function "sqrt" : YES 
Checking for function "pow" : YES 
Checking for function "lfind" : YES 
Checking for function "setmod" : NO 
Checking if "basic inline" compiles: YES 
Configuring tif_config.h using configuration
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/mesonbuild/mesonmain.py", line 131, in run
    return options.run_func(options)
  File "/usr/lib/python3.8/site-packages/mesonbuild/msetup.py", line 245, in run
    app.generate()
  File "/usr/lib/python3.8/site-packages/mesonbuild/msetup.py", line 159, in generate
    self._generate(env)
  File "/usr/lib/python3.8/site-packages/mesonbuild/msetup.py", line 192, in _generate
    intr.run()
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreter.py", line 4359, in run
    super().run()
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 441, in run
    self.evaluate_codeblock(self.ast, start=1)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 466, in evaluate_codeblock
    raise e
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 459, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 472, in evaluate_statement
    return self.function_call(cur)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 829, in function_call
    return func(node, func_args, self.kwargs_string_keys(kwargs))
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 326, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 212, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreter.py", line 3881, in func_subdir
    self.evaluate_codeblock(codeblock)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 466, in evaluate_codeblock
    raise e
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 459, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 482, in evaluate_statement
    return self.evaluate_if(cur)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 566, in evaluate_if
    self.evaluate_codeblock(node.elseblock)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 466, in evaluate_codeblock
    raise e
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 459, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 482, in evaluate_statement
    return self.evaluate_if(cur)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 566, in evaluate_if
    self.evaluate_codeblock(node.elseblock)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 466, in evaluate_codeblock
    raise e
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 459, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 472, in evaluate_statement
    return self.function_call(cur)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 829, in function_call
    return func(node, func_args, self.kwargs_string_keys(kwargs))
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 326, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 212, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreter.py", line 3881, in func_subdir
    self.evaluate_codeblock(codeblock)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 466, in evaluate_codeblock
    raise e
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 459, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 472, in evaluate_statement
    return self.function_call(cur)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 829, in function_call
    return func(node, func_args, self.kwargs_string_keys(kwargs))
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 326, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 326, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 326, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  [Previous line repeated 2 more times]
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 212, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreter.py", line 4079, in func_configure_file
    mesonlib.do_conf_file(inputs_abs[0], ofile_abs, conf.held_object,
  File "/usr/lib/python3.8/site-packages/mesonbuild/mesonlib.py", line 964, in do_conf_file
    line, missing = do_replacement(regex, line, variable_format, confdata)
  File "/usr/lib/python3.8/site-packages/mesonbuild/mesonlib.py", line 880, in do_replacement
    assert variable_format == 'meson'
AssertionError
meson setup: ☒
```

**To Reproduce**
https://github.com/legeana/libopenjp2/blob/dab_wrapdb_add_meson_build/thirdparty/libtiff/meson.build#L306

Minimal example (with a copy of the project header):
https://github.com/legeana/libtiff-meson-bug
```
$ meson builddir
The Meson build system
Version: 0.54.0
Source dir: /home/$user/configure_file_bug/sources/openjpeg-2.3.1/thirdparty/libtiff
Build dir: /home/$user/configure_file_bug/sources/openjpeg-2.3.1/thirdparty/libtiff/builddir
Build type: native build
Project name: openjp2
Project version: 2.3.1
C compiler for the host machine: cc (gcc 9.3.0 "cc (Arch Linux 9.3.0-1) 9.3.0")
C linker for the host machine: cc ld.bfd 2.34
C++ compiler for the host machine: c++ (gcc 9.3.0 "c++ (Arch Linux 9.3.0-1) 9.3.0")
C++ linker for the host machine: c++ ld.bfd 2.34
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (1.6.3)
Run-time dependency zlib found: YES 1.2.11
Run-time dependency libjpeg found: YES 2.0.4
Has header "assert.h" : YES 
Has header "ctype.h" : YES 
Has header "dlfcn.h" : YES 
Has header "fcntl.h" : YES 
Has header "float.h" : YES 
Has header "inttypes.h" : YES 
Has header "ieeefp.h" : NO 
Has header "io.h" : NO 
Has header "limits.h" : YES 
Has header "malloc.h" : YES 
Has header "math.h" : YES 
Has header "memory.h" : YES 
Has header "search.h" : YES 
Has header "stdarg.h" : YES 
Has header "stdint.h" : YES 
Has header "stdlib.h" : YES 
Has header "stdio.h" : YES 
Has header "string.h" : YES 
Has header "strings.h" : YES 
Has header "sys/stat.h" : YES 
Has header "sys/time.h" : YES 
Has header "sys/types.h" : YES 
Has header "time.h" : YES 
Has header "unistd.h" : YES 
Has header "windows.h" : NO 
Has header "winsock2.h" : NO 
Checking for function "getopt" : YES 
Checking for function "isascii" : YES 
Checking for function "memmove" : YES 
Checking for function "memset" : YES 
Checking for function "mmap" : YES 
Checking for function "strcasecmp" : YES 
Checking for function "strchr" : YES 
Checking for function "strrchr" : YES 
Checking for function "strstr" : YES 
Checking for function "strtol" : YES 
Checking for function "strtoul" : YES 
Checking for function "strtoull" : YES 
Checking for function "snprintf" : YES 
Checking for size of "signed int" : 4
Checking for size of "signed long" : 8
Checking for size of "signed long long" : 8
Checking for size of "signed short" : 2
Checking for size of "unsigned int" : 4
Checking for size of "unsigned long" : 8
Checking for size of "unsigned long long" : 8
Checking for size of "unsigned short" : 2
Checking for size of "unsigned char*" : 8
Checking for size of "ssize_t" : 8
Checking for function "floor" : YES 
Checking for function "sqrt" : YES 
Checking for function "pow" : YES 
Checking for function "lfind" : YES 
Checking for function "setmod" : NO 
Checking if "basic inline" compiles: YES 
Configuring tif_config.h using configuration
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/mesonbuild/mesonmain.py", line 131, in run
    return options.run_func(options)
  File "/usr/lib/python3.8/site-packages/mesonbuild/msetup.py", line 245, in run
    app.generate()
  File "/usr/lib/python3.8/site-packages/mesonbuild/msetup.py", line 159, in generate
    self._generate(env)
  File "/usr/lib/python3.8/site-packages/mesonbuild/msetup.py", line 192, in _generate
    intr.run()
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreter.py", line 4359, in run
    super().run()
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 441, in run
    self.evaluate_codeblock(self.ast, start=1)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 466, in evaluate_codeblock
    raise e
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 459, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 472, in evaluate_statement
    return self.function_call(cur)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 829, in function_call
    return func(node, func_args, self.kwargs_string_keys(kwargs))
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 326, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 326, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 326, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  [Previous line repeated 2 more times]
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreterbase.py", line 212, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.8/site-packages/mesonbuild/interpreter.py", line 4079, in func_configure_file
    mesonlib.do_conf_file(inputs_abs[0], ofile_abs, conf.held_object,
  File "/usr/lib/python3.8/site-packages/mesonbuild/mesonlib.py", line 964, in do_conf_file
    line, missing = do_replacement(regex, line, variable_format, confdata)
  File "/usr/lib/python3.8/site-packages/mesonbuild/mesonlib.py", line 880, in do_replacement
    assert variable_format == 'meson'
AssertionError
```

**Expected behavior**
No assertion, an error message or success instead.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)?
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.)
ArchLinux
* what Python version are you using e.g. 3.8.0
```
$ python --version
Python 3.8.2
```
* what `meson --version`
```
$ meson --version
0.54.0
```

**Comment by mensinda** [comment]
Should already be fixed with #6917. Closing

**Comment by dabrain34** [comment]
I confirm :)
